python/ray/experimental/data/datasource/binary_datasource.py

```diff
@@ -0,0 +1,26 @@
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    import pyarrow
+
+from ray.experimental.data.datasource.file_based_datasource import (
+    FileBasedDatasource)
+
+
+class BinaryDatasource(FileBasedDatasource):
+    """Binary datasource, for reading and writing binary files.
+
+    Examples:
+        >>> source = BinaryDatasource()
+        >>> ray.data.read_datasource(source, paths="/path/to/dir").take()
+        ... {"a": 1, "b": "foo"}
+    """
+
+    def _read_file(self, f: "pyarrow.NativeFile", path: str,
+                   **arrow_reader_args):
```

Comment:
nit: it's not arrow_reader args, just reader_args

Reply:
It's currently `arrow_reader_args` for the JSON and CSV datasources, I'll change those as well.

Reply:
Or I can keep those as is, since they're primarily pass-through args to the Arrow readers...